model-optimizer/mo/utils/ir_reader/restore_graph.py

```diff
@@ -17,7 +19,16 @@
 from mo.utils.ir_engine.ir_engine import IREngine
 from mo.utils.ir_reader.layer_to_class import copy_graph_with_ops, collect_extenders, collect_ops
 from mo.utils.utils import get_mo_root_dir
-from extensions.back.MarkNodesWithShapeValues import MarkNodesWithShapeValues
+
+
+def define_data_type(graph: Graph):
+    data_type = 'FP32'
+    for const_node in graph.get_op_nodes(op='Const'):
+        if const_node.soft_get('element_type') == 'f16':
```

Comment:
I think that it is not quite right to detect `f16` nets based on constants. For example, what if we have a model without constants and with floating-point parameter: `Param1->Relu->Result`.

Reply:
Please add a test case for fp16 model without `fp16` constants.

Reply:
Data type detecting was updated, now we use both `Const` and `Parameter` operations to detect. Tests were updated too.